saleor/graphql/product/bulk_mutations/product_variant_bulk_create.py

```diff
@@ -622,28 +662,32 @@ def clean_variants(cls, info, variants, product, errors, index_error_map):
                 "channel"
             ).filter(product=product.id)
         }
-        used_attribute_values = get_used_variants_attribute_values(product)
+
+        variant_attributes = product_type.variant_attributes.annotate(
+            variant_selection=F("attributevariant__variant_selection")
+        )
         variant_attributes_ids = {
-            graphene.Node.to_global_id("Attribute", attr_id)
-            for attr_id in list(
-                product.product_type.variant_attributes.all().values_list(
-                    "pk", flat=True
-                )
-            )
+            graphene.Node.to_global_id("Attribute", variant_attribute.id)
+            for variant_attribute in variant_attributes
         }
+        used_attribute_values = get_used_variants_attribute_values(product)
+
         duplicated_sku = get_duplicated_values(
             [variant.sku for variant in variants if variant.sku]
         )
+        currency_fractions = get_global("currency_fractions")
```

Comment:
Could we load it as global module variable? I think we wouldn't have to pass it around in so many functions.